api/api_test.go

```diff
@@ -88,6 +136,167 @@ func TestAddAlerts(t *testing.T) {
 	}
 }
 
+func TestListAlerts(t *testing.T) {
+	now := time.Now()
+	alerts := []*types.Alert{
+		&types.Alert{
+			Alert: model.Alert{
+				Labels:   model.LabelSet{"state": "active", "alertname": "alert1"},
+				StartsAt: now.Add(-time.Minute),
+			},
+		},
+		&types.Alert{
+			Alert: model.Alert{
+				Labels:   model.LabelSet{"state": "unprocessed", "alertname": "alert2"},
+				StartsAt: now.Add(-time.Minute),
+			},
+		},
+		&types.Alert{
+			Alert: model.Alert{
+				Labels:   model.LabelSet{"state": "suppressed", "silenced_by": "abc", "alertname": "alert3"},
+				StartsAt: now.Add(-time.Minute),
+			},
+		},
+		&types.Alert{
+			Alert: model.Alert{
+				Labels:   model.LabelSet{"state": "suppressed", "inhibited_by": "abc", "alertname": "alert4"},
+				StartsAt: now.Add(-time.Minute),
+			},
+		},
+		&types.Alert{
+			Alert: model.Alert{
+				Labels:   model.LabelSet{"alertname": "alert5"},
+				StartsAt: now.Add(-2 * time.Minute),
+				EndsAt:   now.Add(-time.Minute),
+			},
+		},
+	}
+
+	for i, tc := range []struct {
+		err    bool
+		params map[string]string
+
+		code   int
+		anames []string
+	}{
+		{
+			false,
+			map[string]string{},
+			200,
+			[]string{"alert1", "alert2", "alert3", "alert4"},
+		},
+		{
+			false,
+			map[string]string{"active": "true", "unprocessed": "true", "silenced": "true", "inhibited": "true"},
+			200,
+			[]string{"alert1", "alert2", "alert3", "alert4"},
+		},
+		{
+			false,
+			map[string]string{"active": "false", "unprocessed": "true", "silenced": "true", "inhibited": "true"},
+			200,
+			[]string{"alert2", "alert3", "alert4"},
+		},
+		{
+			false,
+			map[string]string{"active": "true", "unprocessed": "false", "silenced": "true", "inhibited": "true"},
+			200,
+			[]string{"alert1", "alert3", "alert4"},
+		},
+		{
+			false,
+			map[string]string{"active": "true", "unprocessed": "true", "silenced": "false", "inhibited": "true"},
+			200,
+			[]string{"alert1", "alert2", "alert4"},
+		},
+		{
+			false,
+			map[string]string{"active": "true", "unprocessed": "true", "silenced": "true", "inhibited": "false"},
+			200,
+			[]string{"alert1", "alert2", "alert3"},
+		},
+		{
+			false,
+			map[string]string{"filter": "{alertname=\"alert3\""},
+			200,
+			[]string{"alert3"},
+		},
+		{
+			false,
+			map[string]string{"filter": "{alertname"},
+			400,
+			[]string{"alert1", "alert2", "alert3", "alert4"},
```

Comment:
this `[]string{}` should probably be empty, since it's not being executed anyway because of the `400`.